src/vm/analysis/arithmetic_checker/tests.rs

```diff
@@ -129,6 +129,22 @@ fn test_variables_fail_arithmetic_check() {
     }
 }
 
+#[test]
+fn test_version_controlled_variables() {
+    // Certain variables are controlled by the ClarityVersion. E.g., in Clarity1, 'tx-sponsor?'
+    // is not a variable, so can be defined over.
+    let ok_tests = ["(define-private (foo) tx-sponsor?)"];
+
+    for contract in ok_tests.iter() {
+        assert_eq!(
+            arithmetic_check(contract, ClarityVersion::Clarity1),
```

Comment:
Wait, why does this pass?  `tx-sponsor?` only exists as a built-in in `Clarity2`, but this is running in `Clarity1`.

Reply:
I dropped this test method in the refactor. However, the answer is that the contract "passes" because "tx-sponsor?" is NOT reserved in Clarity1, so you can define something over it.